src/components/game.js

```diff
@@ -62,141 +111,257 @@ class Game extends React.Component {
   handleKeyDown = (event) => {
     if (event.key === 'ArrowDown' || event.key === 'ArrowUp'
       || event.key === 'ArrowLeft' || event.key === 'ArrowRight') {
-      this.move(event);
+      // eslint-disable-next-line react/no-unused-state
+      this.setState({ movedList: [] }, () => this.move(event));
     }
   }
 
   move = (direction) => {
     const { data } = this.props;
+    const { setPoints } = this.props;
     const { board } = this.state;
 
     class Element {
       constructor(value, isUsed) {
         this.value = value;
-        this.isUsed = isUsed; // is merged
+        // Tile can't be merged twice in one move therefore
+        // we need to mark tile that was already merged
+        this.isUsed = isUsed;
       }
     }
 
     const temporaryBoard = [];
     let i;
     let j;
     let k;
+    let impossibleMove = true;
+    let pointsToAdd = 0;
     for (i = 0; i < data.site.siteMetadata.boardSize; i += 1) {
       temporaryBoard[i] = new Array(data.site.siteMetadata.boardSize);
     }
 
     for (i = 0; i < data.site.siteMetadata.boardSize; i += 1) {
-      // console.log(`${i}:  `, board[i][0], board[i][1], board[i][2], board[i][3]);
       for (j = 0; j < data.site.siteMetadata.boardSize; j += 1) {
         temporaryBoard[i][j] = new Element(board[i][j], false);
       }
     }
 
-    const merge = (tileMergeIntoPosX, tileMergedIntoPosY,
+    const newMovedList = [];
+
+    const mergeTiles = (tileMergeIntoPosX, tileMergedIntoPosY,
       tileMergedFromPosX, tileMergedFromPosY) => {
       temporaryBoard[tileMergeIntoPosX][tileMergedIntoPosY].value *= 2;
       temporaryBoard[tileMergeIntoPosX][tileMergedIntoPosY].isUsed = true;
       temporaryBoard[tileMergedFromPosX][tileMergedFromPosY].value = 0;
+      newMovedList[newMovedList.length - 1]
+        .toCoords = new Coordinates(tileMergeIntoPosX, tileMergedIntoPosY);
+      newMovedList[newMovedList.length - 1]
```

Comment:
nie chce mi sie tu myśleć ufam że okej i optymalnie to zrobiłeś XD

Reply:
xD mi się wydaje że jest ok